pkg/mimirtool/client/client.go

```diff
@@ -174,36 +173,34 @@ func checkResponse(r *http.Response) error {
 		return nil
 	}
 
-	var msg, errMsg string
-	scanner := bufio.NewScanner(io.LimitReader(r.Body, 512))
-	if scanner.Scan() {
-		msg = scanner.Text()
+	bodyHead, err := io.ReadAll(io.LimitReader(r.Body, 1024))
+	if err != nil {
+		return errors.Wrapf(err, "reading body")
 	}
 
+	msg := string(bodyHead)
+	var errMsg string
 	if msg == "" {
-		errMsg = fmt.Sprintf("server returned HTTP status %s", r.Status)
+		errMsg = fmt.Sprintf("server returned HTTP status: %s", r.Status)
 	} else {
-		errMsg = fmt.Sprintf("server returned HTTP status %s: %s", r.Status, msg)
+		errMsg = fmt.Sprintf("server returned HTTP status: %s, body: %s", r.Status, msg)
```

Comment:
Maybe this should be `%q`, so any funny characters are visible?
I can see arguments on both sides whether newline or `\n` is easier for the reader, but a `\t` would definitely be clearer than some whitespace.